### PR TITLE
RemoteMachineSet controller

### PR DESCRIPTION
**Pull request by abutcher**

### CODE REVIEW

File: pkg/controller/remotemachineset/remotemachineset_controller.go [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+	return client.New(cfg, client.Options{
+		Scheme: scheme,
+	})
+}
+
+func (r *ReconcileRemoteMachineSet) generateInstallConfigFromClusterDeployment(cd *hivev1.ClusterDeployment) (*installtypes.InstallConfig, error) {
+	cdLog := log.WithFields(log.Fields{
+		"clusterDeployment": cd.Name,
+		"namespace":         cd.Namespace,
+	})
```
Comment (dgoodwin) [inline pkg/controller/remotemachineset/remotemachineset_controller.go]: Same as above.

File: pkg/controller/remotemachineset/remotemachineset_controller.go [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+		for _, rMS := range remoteMachineSets.Items {
+			if ms.Name == rMS.Name {
+				found = true
+				if *rMS.Spec.Replicas != *ms.Spec.Replicas ||
+					!reflect.DeepEqual(rMS.Spec.Template.Spec.Labels, ms.Spec.Template.Spec.Labels) ||
+					!reflect.DeepEqual(rMS.Spec.Template.Spec.Taints, ms.Spec.Template.Spec.Taints) {
+					// Capture MachineSet.ResourceVersion and increment MachineSet.Generation
+					ms.ResourceVersion = rMS.ResourceVersion
+					ms.Generation = rMS.Generation + 1
+					machineSetsToUpdate = append(machineSetsToUpdate, ms)
```
[the remark this answers is not in the document]
Comment (abutcher) [reply pkg/controller/remotemachineset/remotemachineset_controller.go]: I've changed this to merge what we wanted to change (`ms.ObjectMeta` and `ms.Spec.Replicas`) with the existing object and then update from existing if we changed. This maintains existing annotations and labels.

File: pkg/controller/remotemachineset/remotemachineset_controller.go [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+func Add(mgr manager.Manager) error {
+	return AddToManager(mgr, NewReconciler(mgr))
+}
+
+// NewReconciler returns a new reconcile.Reconciler
+func NewReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileRemoteMachineSet{
+		Client: mgr.GetClient(),
+		scheme: mgr.GetScheme(),
+		logger: log.WithField("controller", controllerName),
```
Comment (abutcher) [inline pkg/controller/remotemachineset/remotemachineset_controller.go]: Controller log field comes from here now.

**Comment by dgoodwin** [comment]
Looks good to me, lets squash and merge.

**Pull request merged**